### DISCUSSION TITLE
connection to google cloudsql with IAM intergrated account

### DISCUSSION THREAD

**Discussion by neskwoe** [apache/airflow#28173] [Q&A]
Below document talked about connection to cloudsql with static account, however, when we have to use cloud IAM intergrated account(which is google gcp service account). it doesn't seems to have reateld documents, is it possible to do so as of now?

https://airflow.apache.org/docs/apache-airflow-providers-google/stable/connections/gcp_sql.html

**Comment by potiuk** [comment]
I assume this is what you are talking about CloudSQLExecuteQueryOperator - Google Cloud SQL has many operators, most of them are used to manage Cloud SQL instance, but the conneciton you referred to are only used in CloudSQLExecuteQueryOperator to actually run queries (this is described on the page you were referring to):

Google Cloud has many ways you can authenticate to run Cloud SQL "query" .

Have you seen the Howto?

https://airflow.apache.org/docs/apache-airflow-providers-google/stable/operators/cloud/cloud_sql.html#cloudsqlexecutequeryoperator 

It has many examples of connecting via various mechanisms. See if you can find one that fits your case. If not, you need to explain what is your case - ideally pointing out to Google Cloud SQL documentation explaining the authentication model you would like to use.

If in doubt- Google documentation is the most important and source of the truth, so if you point out to the mechanism you want to use and cannot find it in the examples, that would clarify what exactly you mean by using "cloud IAM integrated account".

**Reply by jitendra390** [reply]
@neskwoe  were you able to figure out any solution for this . Im also in the same boat. My customer only allows IAM based authentication and current cloudsql operator only supports DB user based authentication .

**Reply by jitendra390** [reply]
Currently the cloudsql operators from airflow does spin up cloud auth proxy to connect to cloudsql . But it uses DB user name and password to authenticate and authorize.  Cloudsql Auth proxy supports IAM based authentication where the native cloud identity(service account) can be used to authenticate against cloudsql without the need of creating and maintaining a sepearte DB identity. It would be great if the existing cloudsql airflow operators can add this functionality.
